### ISSUE TITLE
Mailer timeout

### ISSUE THREAD

**Issue by seddighi78** [symfony/symfony#35138]
**Symfony version(s) affected**: 4.4.2

**Description**  
I'm using Mailer component for send email in Symfony and set env variables for MAILER_DSN

```
 MAILER_USERNAME=mailer@example.con
 MAILER_PASSWORD=**************
 MAILER_DSN=smtp://$MAILER_USERNAME:$MAILER_PASSWORD@mail.example.com:465
```
the emails was send but throw error for timeout:
```
request.CRITICAL: Uncaught PHP Exception Symfony\Component\Mailer\Exception\TransportException: "Connection to "ssl://mail.example.com:465" timed out." at path/path/vendor/symfony/mailer/Transport/Smtp/Stream/AbstractStream.php line 81 {"exception":"[object] (Symfony\Component\Mailer\Exception\TransportException(code: 0): Connection to \"ssl://mail.grandsocial.net:465\" timed out. at path/path/vendor/symfony/mailer/Transport/Smtp/Stream/AbstractStream.php:81)"} []
```
i think this is bug or if not please help to solve this.
thanks

**Comment by michaljusiega** [comment]
This is not bug. The timeout is default 5 second defined in:
`SocketStream::$timeout = 5;`

There is no configuration here to set this time nor any setter. Increase value anyway.

**Comment by michaljusiega** [comment]
@seddighi78 can you confirm temporary solution ?

**Comment by seddighi78** [comment]
@michaljusiega Please see this comment https://github.com/FriendsOfSymfony/FOSRestBundle/pull/2036#issuecomment-569602695

**Comment by seddighi78** [comment]
@michaljusiega I'm using a try-catch to control the throws of the mailer. the mail was successfully sent but time out happened.

**Comment by michaljusiega** [comment]
Yes because your SMTP server probably analyze your mail too slowly so you got a exception for this reason.

**Comment by seddighi78** [comment]
Do I have to wait for this problem or is there a right solution?

**Comment by azjezz** [comment]
@seddighi78 this is not a bug, as @michaljusiega has mentioned, your server is taking too long to responed and symfony mailer is timing out.

if you want to increase the timeout, you can do so using : 

```php
$transport->getStream()->setTimeout($timeoutInSeconds);
```

**Comment by seddighi78** [comment]
How can i change timeout from event listener or controller? I can't change timeout from configuration

**Comment by fabpot** [comment]
I would highly recommend to fix the latency of the server if possible. If not, configure `default_socket_timeout` in the `php.ini` file (which will be available for SMTP server as of Symfony 5.1). In the meantime, using the snippet given by @azjezz will work as well.

**Comment by michaljusiega** [comment]
Thank you @fabpot

**Comment by leonelvsc** [comment]
The only way of solving this in 4.4 is creating a custom factory that configures the timeout of the socket ?

**Comment by apphancer** [comment]
Here's my problem with this: Application is running on Symfony 5.0 and files between 3-4 MB need to get attached. In the development environment, we are using Mailhog on a Docker container - the former being notoriously slow at processing emails with attachments. In this scenario, we are getting a `["/app/vendor/symfony/mailer/Transport/Smtp/Stream/AbstractStream.php","Connection to "localhost:1025" timed out."]` error from our Console command. Emails are still being sent and received successfully. 
`default_socket_timeout` is set to 60 in the ini settings used by PHP CLI, but this *seems* to be ignored.

**Comment by karser** [comment]
I solved it for v4.4 by setting timeout in the decorated EsmtpTransportFactory::create method.
I [blogged](https://karser.dev/symfony-mailer-increase-socketstream-timeout/) about it.
